src/js/editor/key-commands.js

```diff
@@ -47,13 +47,20 @@ export const DEFAULT_KEY_COMMANDS = [{
 }];
 
 export function validateKeyCommand(keyCommand) {
-  return !!keyCommand.modifier && !!keyCommand.str && !!keyCommand.run;
+  if (!keyCommand.run) {
+    return false;
+  }
+  return !!keyCommand.check || (!!keyCommand.modifier && !!keyCommand.str);
 }
 
 export function findKeyCommand(keyCommands, keyEvent) {
   const key = Key.fromEvent(keyEvent);
 
-  return detect(keyCommands, ({modifier, str}) => {
-    return key.hasModifier(modifier) && key.isChar(str);
+  return detect(keyCommands, ({modifier, str, check}) => {
+    if (!!check) {
+      return check(keyEvent);
```

Comment:
Also as mentioned in https://github.com/bustlelabs/content-kit-editor/issues/168, this could/should also receive the `editor` so decisions could be made as to whether to handle the event based on the current editor state.